### PR TITLE
Call set_avatar_url with target_user, not user_id

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes https://github.com/matrix-org/synapse/issues/8871

### CODE REVIEW

File: synapse/rest/admin/users.py [hunk 1]
```diff
@@ -322,7 +322,7 @@ async def on_PUT(self, request, user_id):
 
             if "avatar_url" in body and type(body["avatar_url"]) == str:
```
Comment (clokep) [inline synapse/rest/admin/users.py:323]: While we're here...

```suggestion
            if "avatar_url" in body and isinstance(body["avatar_url"], str):
```

Comment (Half-Shot) [reply synapse/rest/admin/users.py:323]: ah, good point :)

**Review by clokep — APPROVED** [review]
LGTM if tests pass. It would be nice to add an additional test for this near: https://github.com/matrix-org/synapse/blob/3f0ff53158cc07b481c701077357d9d09254845b/tests/rest/admin/test_user.py#L551

**Comment by Half-Shot** [comment]
Will make an attempt at tests when I have a sec :)

File: changelog.d/8872.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix `PUT /_synapse/admin/v2/users/<user_id>` failing to create a new user when `avatar_url` is specified.
```
Comment (anoadragon453) [inline changelog.d/8872.bugfix:1]:
```suggestion
Fix a bug where `PUT /_synapse/admin/v2/users/<user_id>` failed to create a new user when `avatar_url` is specified. Bug introduced in Synapse v1.9.0.
```

**Review by anoadragon453 — APPROVED** [review]
Change looks good, and can confirm that the new tests fail before this change.

FTR the bug was introduced in https://github.com/matrix-org/synapse/pull/6495.

File: changelog.d/8872.bugfix (hunk 2 above)
Comment (Half-Shot) [reply changelog.d/8872.bugfix]: Thanks for sleuthing

**Pull request merged**